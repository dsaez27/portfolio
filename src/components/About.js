import React from 'react';
import styled from 'styled-components';
import { portfolioImages } from './../helpers/helperImages';

export const About = () => {
    return (
        <>
            <h2>Sobre mí</h2>
            <AboutContainer>
                <AboutCard>
                    <ProfileBackgroundImage
                        src={portfolioImages('./astronout.png').default}
                        alt='Astronout'
                    />
                    <ProfileImage
                        src={portfolioImages('./profile.png').default}
                        alt='Profile Image'
                    />
                    <Profile>
                        <span>Hola, soy Daniel Sáez analista programador</span>
                    </Profile>
                </AboutCard>
            </AboutContainer>
        </>
    );
};

const AboutContainer = styled.div`
    width: 100%;
    height: 100%;
    padding: 2rem 0;
`;

const AboutCard = styled.div`
    display: flex;
    flex-direction: column;
    align-items: center;
    background-color: #2b2d41;
    border-radius: 0.5rem;
    box-shadow: 0px 14px 32px 0px rgb(0 0 0 / 15%);
`;

const ProfileBackgroundImage = styled.img`
    width: 100%;
    height: 200px;
    object-fit: cover;
    border-radius: 0.5rem 0.5rem 0 0;
`;

const ProfileImage = styled.img`
    width: 150px;
    height: 150px;
    object-fit: cover;
    border-radius: 50%;
    transform: translateY(-50%);

    @media (max-width: 768px) {
        width: 100px;
        height: 100px;
    }
`;

const Profile = styled.div`
    width: 100%;
    height: 100%;
    transform: translateY(-50%);
    padding: 1 rem;
    span {
        padding: 0;
        color: #fff;
        font-weight: 300;
    }
`;
