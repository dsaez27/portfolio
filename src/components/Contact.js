import React from "react";
import styled from "styled-components";
import { useForm } from "react-hook-form";

export const Contact = () => {
  const re =
    /[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?/;

  const {
    register,
    handleSubmit,
    formState: { errors },
  } = useForm();

  const onSubmit = data => console.log(data);

  return (
    <div>
      <h1>Contacto</h1>
      <Form onSubmit={handleSubmit(onSubmit)}>
        <Email
          type="text"
          placeholder="Email"
          name="email"
          {...register("email", {
            required: true,
            pattern: re,
          })}
          autoComplete="off"
        />
        <Fields>
          <FieldRequired>* Required</FieldRequired>
          {errors.email?.type === "pattern" && (
            <Error>- Por favor ingrese email correctamente.</Error>
          )}
          {errors.email?.type === "required" && (
            <Error>- El campo es requerido.</Error>
          )}
        </Fields>
        <Message
          placeholder="Mensaje"
          name="message"
          {...register("message", {
            required: true,
            max: 100,
            min: 5,
            maxLength: 100,
            minLength: 5,
          })}
          autoComplete="off"
        />
        <Fields>
          <FieldRequired>* Required</FieldRequired>
          {errors.message?.type === "maxLength" && (
            <Error>- Máximo 100 caracteres.</Error>
          )}
          {errors.message?.type === "minLength" && (
            <Error>- Mínimo 5 caracteres.</Error>
          )}
          {errors.message?.type === "required" && (
            <Error>- El campo es requerido.</Error>
          )}
        </Fields>
        <Submit type="submit">Enviar</Submit>
      </Form>
    </div>
  );
};

const Form = styled.form`
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  justify-content: center;
  width: 500px;
  height: 100%;
  padding: 2rem;

  @media (max-width: 768px) {
    padding: 1rem;
    width: 100%;
  }
`;

const Email = styled.input`
  width: 100%;
  height: 40px;
  border: none;
  padding: 0 10px;
  outline: none;
  margin-bottom: 10px;
  background-color: #2b2d41;
  color: #fff;
`;

const Message = styled.textarea`
  width: 100%;
  height: 100px;
  border: none;
  padding: 0 10px;
  margin-bottom: 10px;
  outline: none;
  overflow: hidden;
  resize: none;
  background-color: #2b2d41;
  font-family: "Open Sans", sans-serif;
  color: #fff;
`;

const Submit = styled.button`
  height: 40px;
  padding: 0 10px;
  width: 200px;

  @media (max-width: 768px) {
    width: 100%;
  }
`;

const Error = styled.p`
  color: #ff4a57;
  font-size: 12px;
  padding: 0;
`;

const FieldRequired = styled.p`
  font-size: 12px;
  padding: 0;
`;

const Fields = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  width: 100%;
  margin-bottom: 5px;
`;
