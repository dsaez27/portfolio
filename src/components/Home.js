import React from 'react';
import Particles from 'react-tsparticles';
import styled from 'styled-components';

import { particles } from './ui/particles';
import './ui/particles.css';
import { Draw } from './ui/Draw';

export const Home = () => {
    return (
        <>
            <Particles
                id='tsparticles'
                options={particles}
                style={{ padding: '0' }}
            />
            <Draw style={{ zIndex: '1' }} />
            <HomeContainer style={{ zIndex: '2' }}>
                <HomeTitle>
                    Hello I'm a <br />
                    Front-end Developer
                </HomeTitle>
                <HomeSubTitle>
                    Fond of creating web application desings and bring them to
                    life using code & develop mobile desings.
                </HomeSubTitle>
                <HomeButton>Creator journey</HomeButton>
            </HomeContainer>
        </>
    );
};

const HomeContainer = styled.div`
    display: flex;
    flex-wrap: nowrap;
    flex-direction: column;
    justify-content: center;
    height: 100vh;
    padding: 2rem 0;
    width: 100%;
    @media (max-width: 768px) {
        justify-content: flex-end;
        padding: 0 0 2rem 0;
    }
`;

const HomeTitle = styled.h1`
    color: #fff;
    @media (max-width: 768px) {
        font-size: 1.5rem;
    }
`;

const HomeSubTitle = styled.p`
    font-size: 1rem;
    padding: 0.5rem 0;
    width: 400px;

    @media (max-width: 530px) {
        width: 90%;
        font-size: 0.9rem;
    }
`;

const HomeButton = styled.button`
    color: #fff;
    height: 3rem;
    width: 10rem;
`;
