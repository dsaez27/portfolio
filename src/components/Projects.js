import React from 'react';
import styled from 'styled-components';
import { Card } from './ui/Card';
import { helperImages } from '../helpers/helperImages';
import { projects } from './../data/projects';

export const Projects = () => {
    return (
        <div>
            <div id='projects' />
            <h2 className='title_section'>Proyectos</h2>
            <CardContainer>
                {projects.map((proyect, index) => {
                    return (
                        <Card
                            key={index}
                            title={proyect.title}
                            description={proyect.description}
                            image={helperImages(proyect.image).default}
                            link={proyect.link}
                            technologies={proyect.technologies}
                        />
                    );
                })}
            </CardContainer>
        </div>
    );
};

const CardContainer = styled.div`
    grid-template-columns: repeat(auto-fit, minmax(260px, 1fr));
    display: grid;
    grid-gap: 1rem;
`;
