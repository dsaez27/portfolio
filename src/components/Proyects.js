import React from 'react';
import styled from 'styled-components';
import { proyects } from '../data/proyects';
import { Card } from './ui/Card';
import { portfolioImages } from './../helpers/helperImages';

export const Proyects = () => {
    return (
        <div>
            <h2>Proyectos</h2>

            <CardContainer>
                {proyects.map((proyect, index) => {
                    return (
                        <Card
                            key={index}
                            title={proyect.title}
                            description={proyect.description}
                            image={portfolioImages(proyect.image).default}
                            link={proyect.link}
                            technologies={proyect.technologies}
                        />
                    );
                })}
            </CardContainer>
        </div>
    );
};

const CardContainer = styled.div`
    grid-template-columns: repeat(auto-fit, minmax(260px, 1fr));
    display: grid;
    grid-gap: 1.2rem;
    padding: 2rem 0;
`;
