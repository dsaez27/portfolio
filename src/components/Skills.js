import React from 'react';
import styled from 'styled-components';
import { portfolioImages } from './../helpers/helperImages';
import { skills } from './../data/Skills';

export const Skills = () => {
    return (
        <div>
            <h2>Skills</h2>
            <SkillsContainer>
                {skills.map((skill, index) => {
                    return (
                        <SkillItem key={index}>
                            <SkillIcon
                                src={portfolioImages(skill.image).default}
                                alt={skill.name}
                            />
                            <SkillName>{skill.name}</SkillName>
                        </SkillItem>
                    );
                })}
            </SkillsContainer>
        </div>
    );
};

const SkillsContainer = styled.div`
    display: grid;
    grid-gap: 1.5rem;
    grid-template-columns: repeat(auto-fit, minmax(150px, 1fr));
    padding: 2rem 0;
`;

const SkillItem = styled.div`
    display: flex;
    flex-direction: column;
    align-items: center;
    background-color: #2b2d41;
    box-shadow: 0px 14px 32px 0px rgba(0, 0, 0, 0.15);
    padding: 10px;
    border-radius: 3px;
`;

const SkillIcon = styled.img`
    width: 2rem;
    height: 2rem;
    margin: 1rem;
`;

const SkillName = styled.p`
    font-size: 0.8rem;
    padding: 0.5rem;
    cursor: default;
`;
