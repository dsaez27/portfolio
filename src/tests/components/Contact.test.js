import { act, render } from "@testing-library/react";
import { mount } from "enzyme";
import { Contact } from "./../../components/Contact";

describe("Pruebas en <Contact />", () => {
  const wrapper = mount(<Contact />);

  test("Debe de mostrarse corrctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de mostrar error si no se pasa un email", () => {
    act(() => {
      render(<Contact />, wrapper);
    });
    
    expect(wrapper.find("p").at(2).text()).toEqual("Error");
  });
});
