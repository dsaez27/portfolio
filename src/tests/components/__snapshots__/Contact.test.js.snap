// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Contact /> Debe de mostrarse correctamente 1`] = `
Array [
  <h2>
    Contacto
  </h2>,
  <div
    className="sc-furwcr bEGSHh"
  >
    <div>
      <form
        id="form-validation"
        onSubmit={[Function]}
      >
        <label
          htmlFor="name"
        >
          Nombre
        </label>
        <input
          id="name"
          name="name"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Nombre y apellido"
          type="text"
        />
        <p />
        <label
          htmlFor="email"
        >
          Email
        </label>
        <input
          id="email"
          name="email"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Email"
          type="email"
        />
        <p />
        <label
          htmlFor="message"
        >
          Mensaje
        </label>
        <textarea
          cols="50"
          id="message"
          name="message"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Mensaje"
          rows="10"
        />
        <p />
        <button
          type="submit"
        >
          Enviar
        </button>
      </form>
    </div>
  </div>,
]
`;
